packages/date-time-picker/src/DateTimePicker.ts

```diff
@@ -0,0 +1,159 @@
+governing permissions and limitations under the License.
+*/
+import {
+    CSSResultArray,
+    html,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import {
+    property,
+    state,
+} from '@spectrum-web-components/base/src/decorators.js';
+import { InputSegments } from '@spectrum-web-components/input-segments';
+
+import styles from './date-time-picker.css.js';
+
+// TODO: Load dependencies lazily when possible
+import '@spectrum-web-components/calendar/sp-calendar.js';
+import '@spectrum-web-components/icons-workflow/icons/sp-icon-calendar.js';
+import '@spectrum-web-components/overlay/sp-overlay.js';
+import '@spectrum-web-components/picker-button/sp-picker-button.js';
+import '@spectrum-web-components/popover/sp-popover.js';
+
+/**
+ * @element sp-date-time-picker
+ *
+ * @event change - Announces when a new date/time is defined by emitting a `Date` object
+ *
+ * @slot calendar-icon - The icon used in the calendar button
+ * @slot help-text - Default or non-negative help text to associate to your form element
+ * @slot negative-help-text - Negative help text to associate to your form element when `invalid`
+ */
+export class DateTimePicker extends InputSegments {
+    public static override get styles(): CSSResultArray {
+        return [...super.styles, styles];
+    }
+
+    /**
+     * Indicates whether the picker should be displayed or not
+     */
+    @property({ type: Boolean, reflect: true })
+    public open = false;
+
+    @state()
+    override includeDate = true;
+
+    @state()
+    override includeTime = true;
+
+    @state()
+    private pickerDate?: Date;
+
+    protected override renderField(): TemplateResult {
+        return html`
+            ${this.renderInputContent()} ${this.renderPicker()}
+        `;
+    }
+
+    public renderPicker(): TemplateResult {
+        const isDisabled = this.disabled || this.readonly;
+
+        return html`
+            <div class="picker">
+                ${this.renderStateIcons()}
+
+                <sp-picker-button
+                    class="picker-button"
+                    ?open=${this.open}
+                    ?quiet=${this.quiet || this.readonly}
+                    ?invalid=${this.invalid}
+                    ?disabled=${isDisabled}
+                    @click=${this.showPicker}
+                >
+                    <slot name="calendar-icon" slot="icon">
+                        <sp-icon-calendar></sp-icon-calendar>
+                    </slot>
+                </sp-picker-button>
+
+                <!-- TODO: Enable "receives-focus" when calendar is navigable via keyboard -->
+                <sp-overlay
+                    placement="bottom-start"
+                    receives-focus="false"
+                    .triggerElement=${this as HTMLElement}
+                    ?open=${this.open}
+                    @sp-closed=${this.hidePicker}
+                >
+                    <sp-popover class="popover">
```

Comment:
FYI: This creates a stringent boundary when we would want to render this in mobile!
In mobile the calender should render in a tray, not sure if it has been design approved or not!

Reply:
AFAIK there's no design for mobile. If the spectrum design team has something specific in mind we can do that. I agree that aligning with the React Spectrum approach (using the tray) should be desired but for the moment this approach can work. I can change this in a future PR before release and I think the MatchMediaController can be used here.

Reply:
Right! This is just na FYI